docs/src/modules/concepts/pages/ai-agents.adoc

```diff
@@ -0,0 +1,62 @@
+Answer the question based on the context below. 
+Keep the answer short and concise. Respond "Unsure about answer"
+if not sure about the answer.
+
+If asked for a single item and multiple pieces of information 
+are acceptable answers, choose one at random.
+
+Context:
+ Here is a summary of all the action movies you know of. Each one is rated from 1 to 5 stars.
+
+Question:
+  What is the most highly rated action movie?
+----
+
+Everything except the **question** above would have been supplied by the agent. Working with and honing prompts is such an important activity in agentic development that a whole new discipline called https://www.promptingguide.ai/[prompt engineering] has sprung up around it.
+
+The context in the preceding prompt is how agents can augment the knowledge of an LLM. This is how Retrieval Augmented Generation (RAG) works. Agents can participant in sessions where the conversation history is stored. You see this in action whenever you use an AI assistant and it shows you a history of all of your chats. Session management and persistence is a task every agent developer needs to tackle.
+
+
+== Agent orchestration and collaboration
+
+Each agent should do _one thing_. Agents should have a single goal and they can use any form of knowledge and model inference to accomplish that goal. However, we rarely ever build applications that only do one thing. One of the super powers of agents is in _collaboration_.
+
+There are protocols and standards rapidly evolving for ways agents can communicate with each other directly, but agents also benefit from indirect communciation. 
+
+Whether you have 1 agent or 50, you still need to be able to do things like recover from network failure, handle timeouts, deal with failure responses, broken streams, and much more. Just for individual agents you will need an orchestrator if you want that agent to be resilient at scale. With dozens of agents working together with shared and isolated sessions, they will need to be managed by supervisors.
+
+== Agent evaluation
+
+The answers your agents get from models are _non-deterministic_. They can seem random at times. Since you can't predict the model output you can't use traditional testing practices. Instead, we need to do what's called **evaluation**. Evaluation involves iteratively refining a prompt.
+
+You submit the prompt to the model and get an answer back. Then, we can use _another model_ to derive metrics from that response like confidence ratings. This is often called the "LLM-as-judge" pattern and is becoming pretty popular. 
+
+Rather than a unit test, we often have entire suites of evaluation runs where we submit a large number of prompts, derive analytics and metrics from the replies, and then score the model-generated data as a whole. This is tricky because you can very easily have an evaluation run that has a high confidence score but still somehow manages to contain https://www.ibm.com/think/topics/ai-hallucinations[hallucinations].
```

Comment:
```suggestion
Rather than a unit test, we often have entire suites of evaluation runs where we submit a large number of prompts, derive analytics and metrics from the replies, and then score the model-generated data as a whole. This is tricky because you can very easily have an evaluation run that has a high confidence score but still somehow manages to contain link:https://www.ibm.com/think/topics/ai-hallucinations[hallucinations].
```